pkg/services/control/ir/service.proto

```diff
@@ -41,3 +50,89 @@ message HealthCheckResponse {
     // Body signature.
     Signature signature = 2;
 }
+
+// NotaryList request.
+message NotaryListRequest {
+    // Request body structure.
+    message Body {
+    }
+
+    // Body of request message.
+    Body body = 1;
+
+    // Body signature.
+    Signature signature = 2;
+}
+
+// NotaryList response.
+message NotaryListResponse {
+    // Request body structure.
+    message Body {
+        // Transaction of the notary request.
+        repeated TransactionInfo transactions = 1;
+    }
+
+    // Body of request message.
+    Body body = 1;
+
+    // Body signature.
+    Signature signature = 2;
+}
+
+// NotaryRequest request.
+message NotaryRequestRequest {
+    // Request body structure.
+    message Body {
+        string method = 1;
+        repeated string args = 2;
+    }
+
+    // Body of request message.
+    Body body = 1;
+
+    // Body signature.
+    Signature signature = 2;
+}
+
+// NotaryRequest response.
+message NotaryRequestResponse {
+    // Request body structure.
+    message Body {
+        // Result hash of transaction.
+        string hash = 1;
+    }
+
+    // Body of request message.
+    Body body = 1;
+
+    // Body signature.
+    Signature signature = 2;
+}
+
+// NotarySign request.
+message NotarySignRequest {
+    // Request body structure.
+    message Body {
+        // Hash of transaction that need to be signed.
+        string hash = 1;
```

Comment:
i think raw representation would be better as we usually prefer in our protobuf protocol cc @roman-khimov